istio-control/istio-autoinject/files/injection-template.yaml

```diff
@@ -0,0 +1,342 @@
+{{- if .Values.global.proxy.resources }}
+    {{ toYaml .Values.global.proxy.resources | indent 4 }}
+{{- end }}
+  {{  end -}}
+  volumeMounts:
+  {{ if (isset .ObjectMeta.Annotations `sidecar.istio.io/bootstrapOverride`) }}
+  - mountPath: /etc/istio/custom-bootstrap
+    name: custom-bootstrap-volume
+  {{- end }}
+  - mountPath: /etc/istio/proxy
+    name: istio-envoy
+  {{- if .Values.global.sds.enabled }}
+  - mountPath: /var/run/sds/uds_path
+    name: sds-uds-path
+    readOnly: true
+  {{- if .Values.global.sds.useTrustworthyJwt }}
+  - mountPath: /var/run/secrets/tokens
+    name: istio-token
+  {{- end }}
+  {{- if .Values.global.sds.customTokenDirectory }}
+  - mountPath: "{{ .Values.global.sds.customTokenDirectory -}}"
+    name: custom-sds-token
+    readOnly: true
+  {{- end }}
+  {{- else }}
+  - mountPath: /etc/certs/
+    name: istio-certs
+    readOnly: true
+  {{- end }}
+  {{- if and (eq .Values.global.proxy.tracer "lightstep") .Values.global.tracer.lightstep.cacertPath }}
+  - mountPath: {{ directory .ProxyConfig.GetTracing.GetLightstep.GetCacertPath }}
+    name: lightstep-certs
+    readOnly: true
+  {{- end }}
+    {{- if isset .ObjectMeta.Annotations `sidecar.istio.io/userVolumeMount` }}
+    {{ range $index, $value := fromJSON (index .ObjectMeta.Annotations `sidecar.istio.io/userVolumeMount`) }}
+  - name: "{{  $index }}"
+    {{ toYaml $value | indent 4 }}
+    {{ end }}
+    {{- end }}
+volumes:
+{{- if (isset .ObjectMeta.Annotations `sidecar.istio.io/bootstrapOverride`) }}
+- name: custom-bootstrap-volume
+  configMap:
+    name: {{ annotation .ObjectMeta `sidecar.istio.io/bootstrapOverride` "" }}
+{{- end }}
+- emptyDir:
+    medium: Memory
+  name: istio-envoy
+{{- if .Values.global.sds.enabled }}
+- name: sds-uds-path
+  hostPath:
+    path: /var/run/sds/uds_path
+    type: Socket
+{{- if .Values.global.sds.customTokenDirectory }}
+- name: custom-sds-token
+  secret:
+    secretName: sdstokensecret
+{{- end }}
+{{- if .Values.global.sds.useTrustworthyJwt }}
+- name: istio-token
+  projected:
+    sources:
+    - serviceAccountToken:
+        path: istio-token
+        expirationSeconds: 43200
+        audience: {{ .Values.global.trustDomain }}
+{{- end }}
+{{- else }}
+- name: istio-certs
+  secret:
+    optional: true
+    {{ if eq .Spec.ServiceAccountName "" }}
+    secretName: istio.default
+    {{ else -}}
+    secretName: {{  printf "istio.%s" .Spec.ServiceAccountName }}
+    {{  end -}}
+  {{- if isset .ObjectMeta.Annotations `sidecar.istio.io/userVolume` }}
+  {{range $index, $value := fromJSON (index .ObjectMeta.Annotations `sidecar.istio.io/userVolume`) }}
+- name: "{{ $index }}"
+  {{ toYaml $value | indent 2 }}
+  {{ end }}
+  {{ end }}
+{{- end }}
+{{- if and (eq .Values.global.proxy.tracer "lightstep") .Values.global.tracer.lightstep.cacertPath }}
+- name: lightstep-certs
+  secret:
+    optional: true
+    secretName: lightstep.cacert
+{{- end }}
+{{- if .Values.global.podDNSSearchNamespaces }}
```

Comment:
this last bit can be commented out.  Most of the implementation for this was reverted in 1.1 - except for the DNS resolution.  IOW, this doesn't work and is likely to cause confusion.  See Issue https://github.com/istio/istio/issues/13045